hppc/src/main/java/com/carrotsearch/hppc/sorting/IndirectSort.java

```diff
@@ -34,7 +34,8 @@ private IndirectSort() {
    * Returns the order of elements between indices <code>start</code> and <code>length</code>, as
    * indicated by the given <code>comparator</code>.
    *
-   * <p>This routine uses merge sort. It is guaranteed to be stable.
+   * <p>This routine uses merge sort. It is guaranteed to be stable. It creates a new indices array,
+   * and clones it while sorting.
    */
   public static int[] mergesort(int start, int length, IntBinaryOperator comparator) {
     final int[] src = createOrderArray(start, length);
```

Comment:
Wait... was it like this before? Becuase that version of mergesort that accepts an external array shouldn't clone it - it doesn't make sense? And right now the mergesort(int, int, comparator) creates a new array just so that it can be cloned in a call to sort(int[], comparator)...